### ISSUE TITLE
Markdown preview ignores line breaks for note title and content

### ISSUE THREAD

**Issue by reginabally** [Automattic/Simplenote-United#108]
Similar to [this issue](https://github.com/Automattic/Simplenote-United/issues/31), I notice the markdown preview ignores the line break after the note title on the iOS, Android, and the Electron app. This issue isn't observed on the macOS app.

### Expected

I would expect a line break will be added right after the note title when I preview the note.

### Observed

The note title and its content are displayed on one single line on the preview.

### Reproduced

1. Create a new note on the Simplenote app with Markdown enabled
2. Type the note title and press Enter once
3. Type some text and press Enter once
4. Type some more text and press Enter once
5. Preview the note and see that all text added from Step 2 to 4, including the note title appears on the same line.

Screen recording taken on the iOS app: https://d.pr/v/nGOZ36/NTbgvykN3H

Screenshot when previewing the note on the Android app:

![Screenshot_2021-02-02-13-32-26-42_dd6b2219cb0a90b8a293feeeaec0487a](https://user-images.githubusercontent.com/40906847/106557059-f7dfca00-655b-11eb-8c62-3c162b492c11.jpg)

Screen recording when previewing on Electron:

![Screen Capture on 2021-02-02 at 13-29-26](https://user-images.githubusercontent.com/40906847/106557082-0928d680-655c-11eb-8a82-307c21843cad.gif)

Screen recording when previewing on macOS app version 2.6:

![Screen Capture on 2021-02-02 at 13-28-51](https://user-images.githubusercontent.com/40906847/106557112-16de5c00-655c-11eb-98fe-e2133c1e7333.gif)

### Where did you see the bug

- System Make: Apple, OPPO
- System Model: iPhone SE, Reno 4
- OS: iOS, Android
- OS version: iOS 14.3, Android 10
- Browser (if applicable): Google Chrome on macOS
- Browser version (if applicable):Version 88.0.4324.96 (Official Build) (x86_64)
- Simplenote app version: 4.30 on iOS, 2.16-rc-2 (132) on Android

Note: It was originally reported in 3670504-zen.

**Comment by codebykat** [comment]
This is technically correct (it's macOS that's in error?) 🤔 

> Ending a line with no spaces or with just one space doesn't create a line break. Use two or more spaces to create a line break.

https://riptutorial.com/markdown/example/5750/line-breaks-and-paragraphs

<img src="https://media.giphy.com/media/1hMk0bfsSrG32Nhd5K/giphy.gif" />

**Comment by reginabally** [comment]
Is it expected to add two spaces after the note title to create the line break? 🤔

**Comment by codebykat** [comment]
Yeah, we don't treat the title specially in Markdown mode, to allow for Markdown formatting to work as expected. The title is just the first line of the note.

This seems to be behaving consistently for me. Tested on web and published note.

* A title and a second line with only a single line break gets converted to one line ✅ 

```
line break testing
second line with no line break is the same line
```
<img width="629" alt="Screen Shot 2021-02-04 at 9 24 46 AM" src="https://user-images.githubusercontent.com/52152/106931251-3942b780-66cb-11eb-9c74-8fcb34fc7cd9.png">
    
* A title with two line breaks (a blank line before the content) produces a new paragraph ✅ 

```
line break testing

second line after a line break creates a new paragraph
```

<img width="579" alt="Screen Shot 2021-02-04 at 9 24 27 AM" src="https://user-images.githubusercontent.com/52152/106931407-6becb000-66cb-11eb-81de-b4bfe174a277.png">

* A title with two spaces at the end of the line produces a line break but not a new paragraph ✅ 

<img width="607" alt="Screen Shot 2021-02-04 at 9 26 09 AM" src="https://user-images.githubusercontent.com/52152/106931171-216b3380-66cb-11eb-9746-ae38fc1ca946.png">

<img width="708" alt="Screen Shot 2021-02-04 at 9 25 50 AM" src="https://user-images.githubusercontent.com/52152/106931203-2af49b80-66cb-11eb-9071-6231f7211219.png">

**Comment by mdrockwell** [comment]
I believe this might be a related issue. User contacted us letting us know that the preview in the app is inconsistent with the published note. Here is the text they sent:

```
This is a paragraph with a list starting right underneath it
- This is a list item
- This is another list item
```

This behaves differently depending on where it is being viewed.

**Preview on Mac, Simplenote native 2.7:**

![Markup on 2021-02-11 at 10:26:46](https://user-images.githubusercontent.com/42454301/107657995-b635da00-6c53-11eb-8fe2-dbe0ebaa7d22.png)

Preview on iOS, Simplenote version **4.32:**

![IMG_4AD7E82E9BD7-1](https://user-images.githubusercontent.com/42454301/107658453-25133300-6c54-11eb-8075-7ecd0f993bf1.jpeg)

**Preview n Android, Simplenote version 2.16-rc-3:**

![Screenshot_20210211-103434](https://user-images.githubusercontent.com/42454301/107659434-24c76780-6c55-11eb-8c55-7ed924e47490.png)

**On the web app:**

![Markup on 2021-02-11 at 10:27:42](https://user-images.githubusercontent.com/42454301/107658097-ce0d5e00-6c53-11eb-89d3-4dc998b509f3.png)

**When published:**

![Markup on 2021-02-11 at 10:28:39](https://user-images.githubusercontent.com/42454301/107658254-f09f7700-6c53-11eb-9325-2c8874acba45.png)

-------------

Adding two spaces at the end of the "This is a paragraph with a list starting right underneath it" text will add a line break on Mac, iOS, and Android, but the markdown list doesn't render as a list.

Ticket: 3730433-zd-woothemes

**Comment by erricgunawan** [comment]
Just noting, when searching for this issue, I found two others that might be related (or the same?):

https://github.com/Automattic/Simplenote-United/issues/105
https://github.com/Automattic/Simplenote-United/issues/137

**Comment by erricgunawan** [comment]
> I noticed following differences between Android app and web/desktop app:
>
> ## Android app
> - in edit mode first line is in bold
> - note list includes `#` character of markdown titles
> - in preview mode new line character is not rendered
>
> ## web app & linux app
> - in edit mode first line is not in bold
> - note list does not include `#` character of markdown titles
> - in preview mode new line character is rendered
>
> From my perspective, web/desktop app behaviour is better than in the case of Android. I would suggest improvements in the later case.

<details>
  <summary>Screenshot from Android:</summary>
  <br>
  
![android-edit-mode png](https://user-images.githubusercontent.com/1813857/182097567-bf1d97bd-2c52-4a27-b2fb-84aa878372c9.png)
![android-preview-mode](https://user-images.githubusercontent.com/1813857/182097584-3fc45f6b-5468-4e39-8522-f9b3146d7aec.png)
![android-list](https://user-images.githubusercontent.com/1813857/182097592-2cd76056-47bb-493a-b998-378a27b35e48.png)

  
</details>

<details>
  <summary>Screenshot from Desktop:</summary>
  <br>
  
![desktop-edit-mode](https://user-images.githubusercontent.com/1813857/182097611-f00fd36f-c302-4047-840b-4e29362f35de.png)
![desktop-preview-mode](https://user-images.githubusercontent.com/1813857/182097647-2b84528f-bb1d-4f59-b7a7-c49bc78ee5a2.png)

  
</details>

5421963-zen
